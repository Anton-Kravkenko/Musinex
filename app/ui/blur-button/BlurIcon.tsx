import UIcon from '@/ui/icon/defaultIcon/Icon'
import { getHexCode } from '@/utils/getColor'
import { BlurView } from 'expo-blur'
import { useColorScheme } from 'nativewind'
import { FC, PropsWithChildren } from 'react'
import { Pressable } from 'react-native'
import { IblurButton } from './types/IblurButton'

const BlurIcon: FC<PropsWithChildren<IblurButton>> =
	({
		 children,
		 color,
		 iconSize = 21,
		 isSmall = false,
		 icon,
		 ...rest
	 }) => {
		const { colorScheme } = useColorScheme()
		return (
			<Pressable {...!rest.onPress && rest}>
				<BlurView
					intensity={22}
					tint='light'
					className={'items-center justify-center overflow-hidden'}
					style={{
						width: isSmall ? 15 : 40,
						height: isSmall ? 15 : 40,
						borderRadius: isSmall ? 4 : 100
					}}
				>
					{(children || !icon) ? (
						children
					) : (
						<UIcon
							type={'ionicon'}
							onPress={rest.onPress}
							name={icon}
							size={iconSize}
							color={color ? color : colorScheme === 'light' ? getHexCode('dark') : getHexCode('white')}
						/>
					)}
				</BlurView>
			</Pressable>
		)
	}

export default BlurIcon
