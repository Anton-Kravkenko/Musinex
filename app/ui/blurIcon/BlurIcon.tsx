import { BlurButtonProps } from '@/ui/blurIcon/blurButton.types'
import Icon from '@/ui/icon/default-icon/icon'
import { Color } from '@/utils/color'
import { BlurView } from 'expo-blur'
import { FC, memo, PropsWithChildren } from 'react'

const BlurIcon: FC<PropsWithChildren<BlurButtonProps>> = ({
	children,
	color = Color.white,
	iconSize = 21,
	isSmall = false,
	icon = 'alert-outline',
	onPress,
	style,
	...props
}) => (
	<BlurView
		intensity={22}
		tint='light'
		testID='blur-icon'
		className='items-center justify-center overflow-hidden'
		style={[
			{
				width: isSmall ? 15 : 40,
				height: isSmall ? 15 : 40,
				borderRadius: isSmall ? 4 : 8
			},
			style
		]}
		{...props}>
		{children || !icon ? (
			children
		) : (
			<Icon
				onPress={onPress}
				name={icon}
				size={iconSize}
				color={color ? color : Color.white}
			/>
		)}
	</BlurView>
)

export default memo(BlurIcon)
