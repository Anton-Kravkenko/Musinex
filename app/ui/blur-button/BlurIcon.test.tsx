import BlurIcon from '@/ui/blur-button/BlurIcon'
import { render } from '@testing-library/react-native'

describe('BlurIcon', () => {
		it('should render correctly', () => {
		const renders = render(<BlurIcon />)
		expect(renders).toMatchSnapshot()
		})
})