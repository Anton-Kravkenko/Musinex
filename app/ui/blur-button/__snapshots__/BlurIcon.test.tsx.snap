// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlurIcon should render correctly 1`] = `
<View
  className="items-center justify-center overflow-hidden"
  style={
    [
      {
        "backgroundColor": "transparent",
      },
      {
        "borderRadius": 8,
        "height": 40,
        "width": 40,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoBlurView
    proxiedProperties={
      {
        "intensity": 22,
        "tint": "light",
      }
    }
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
